From 0d333f647d207d7dbce2fb68b4ff347e2fefaa30 Mon Sep 17 00:00:00 2001
From: =?utf8?q?Morten=20Johan=20S=C3=B8rvig?= <morten.sorvig@qt.io>
Date: Tue, 23 Jan 2018 14:56:40 +0100
Subject: [PATCH 1/1] Make QWebView use the native backend on macOS

Revert back to the original, documented behavior where
QWebView is implemented using a native web-browser
component.

This makes QWebView usable for use cases where deploying
a QWebEngine is not possible, e.g. on the Mac App Store.

The QT_MAC_USE_NATIVE_WEBVIEW environment variable
is no longer read.

Change-Id: I038a66a90ca064e424e026175c02f812e737ea6b
===
 qtwebview.pro                      | 2 +-
 src/webview/qtwebviewfunctions.cpp | 8 +++-----
 src/webview/qwebview_darwin.mm     | 4 ----
 src/webview/webview-lib.pri        | 2 --
 4 files changed, 4 insertions(+), 12 deletions(-)

diff --git a/qtwebview/qtwebview.pro b/qtwebview/qtwebview.pro
index 66de876..828bf9e 100644
--- a/qtwebview/qtwebview.pro
+++ b/qtwebview/qtwebview.pro
@@ -1,3 +1,3 @@
-requires(android|ios|winrt|qtHaveModule(webengine))
+requires(android|ios|macos|winrt|qtHaveModule(webengine))
 
 load(qt_parts)
diff --git a/qtwebview/src/webview/qtwebviewfunctions.cpp b/qtwebview/src/webview/qtwebviewfunctions.cpp
index 7509f21..a98c467 100644
--- a/qtwebview/src/webview/qtwebviewfunctions.cpp
+++ b/qtwebview/src/webview/qtwebviewfunctions.cpp
@@ -75,9 +75,11 @@ void QtWebView::initialize()
         qputenv("QSG_RENDER_LOOP", "basic");
     } else
 #endif
+    {
 #if defined(QT_WEBVIEW_WEBENGINE_BACKEND)
-    QtWebEngine::initialize();
+        QtWebEngine::initialize();
 #endif
+    }
 }
 
 /*!
@@ -87,11 +89,7 @@ void QtWebView::initialize()
 
 bool QtWebViewPrivate::useNativeWebView()
 {
-#ifdef Q_OS_MACOS
-    return qEnvironmentVariableIsSet("QT_MAC_USE_NATIVE_WEBVIEW");
-#else
     return true;
-#endif
 }
 
 QT_END_NAMESPACE
diff --git a/qtwebview/src/webview/qwebview_darwin.mm b/qtwebview/src/webview/qwebview_darwin.mm
index ff9dc4d..36fd79c 100644
--- a/qtwebview/src/webview/qwebview_darwin.mm
+++ b/qtwebview/src/webview/qwebview_darwin.mm
@@ -64,10 +64,6 @@ QT_BEGIN_NAMESPACE
 
 QWebViewPrivate *QWebViewPrivate::create(QWebView *q)
 {
-#ifdef Q_OS_MACOS
-    if (!QtWebViewPrivate::useNativeWebView())
-        return new QWebEngineWebViewPrivate(q);
-#endif
     return new QDarwinWebViewPrivate(q);
 }
 
diff --git a/qtwebview/src/webview/webview-lib.pri b/qtwebview/src/webview/webview-lib.pri
index f5d6774..a802365 100644
--- a/qtwebview/src/webview/webview-lib.pri
+++ b/qtwebview/src/webview/webview-lib.pri
@@ -52,8 +52,6 @@ android {
         qwebview_darwin_p.h
     LIBS_PRIVATE += -framework Foundation -framework WebKit
     ios: LIBS_PRIVATE += -framework UIKit
-
-    macos: CONFIG += use_webengine_backend
 } else: winrt {
     NO_PCH_SOURCES += qwebview_winrt.cpp
     SOURCES += $$COMMON_SOURCES
-- 
2.7.4

